src/bootstrap/lib.rs

```diff
@@ -718,7 +718,7 @@ impl Build {
     fn force_use_stage1(&self, compiler: Compiler, target: Interned<String>) -> bool {
         !self.config.full_bootstrap &&
             compiler.stage >= 2 &&
-            self.hosts.iter().any(|h| *h == target)
+            (self.hosts.iter().any(|h| *h == target) || target == self.build)
```

Comment:
Hm, something is wrong if this is necessary. I believe we try to make the first host `self.build` in `config.rs`; could you check on that logic?

Reply:
Yeah I tried reordering this in `config.rs` but then it ended up building too much b/c it's using `config.host` to seed the matrix of what to build presumably? That is, when we do `--host foo`, we only want the host binaries for `foo` and not for the build triple. Do you think there's a better way to solve this though?

Reply:
Probably not then -- I think our current separation between target/host/build isn't really well thought out and probably various build steps build on the wrong platform (i.e., not optimal time-wise) today. I'm hoping to categorize the list of steps and make the distinction clearer.